terraform/node_module_expand.go

```diff
@@ -182,27 +170,8 @@ func (n *evalCloseModule) Eval(ctx EvalContext) (interface{}, error) {
 			}
 		}
 
-		found := false
-		if n.orphaned {
-			// we're removing the entire module, so all instances must go
-			found = true
-		} else {
-			// if this instance is not in the current expansion, remove it from
-			// the state
-			for _, current := range currentModuleInstances {
-				if current.Equal(mod.Addr) {
-					found = true
-					break
-				}
-			}
-		}
-
-		if !found {
-			if len(mod.Resources) > 0 {
-				// FIXME: add more info to this error
-				return nil, fmt.Errorf("module %q still contains resources in state", mod.Addr)
-			}
-
+		// empty child modules are always removed
+		if len(mod.Resources) == 0 && !mod.Addr.IsRoot() {
```

Comment:
If a child module has no resources, but has outputs, this changeset reflects that those outputs will no longer be output. Is this an expected result of this change? I am intrigued because it seems like this could be a particular breaking change to note if so.

Reply:
This is reflected in the test updates here: https://github.com/hashicorp/terraform/pull/25018/files#diff-62779ef840b004fccd0d20eb74ea741eL444

Reply:
Reopen if I have this wrong but: This isn't an issue because the outputs showing in our tests were a result of in-memory state, and not truly output? As such, I'll resolve this comment but again, please reopen if it's still a question.